pkg/cloudinfo/cloudinfo.go

```diff
@@ -292,6 +316,9 @@ func (cpi *CachingCloudInfo) renewShortLived(ctx context.Context) {
 
 		providerWg.Add(1)
 		go func(c context.Context, p string, i CloudInfoer) {
+			ctxp, _ := cpi.tracer.StartWitTags(ctx, fmt.Sprintf("collect-provider-price-info (%s)", p), map[string]interface{}{"provider": p})
```

Comment:
It is not safe to start a span from a parent in a different goroutine. You should start a new root (remote) span and link to the original trace using trace and span ids.

Reply:
thx, fixed it, though the jaeger ui doesn't seem to properly handle links (the json representation HAS the link)

Reply:
fixed, this one; others to come - links are not properly displayed by the jaeger ui